qubesbuilder/plugins/source/scripts/get-and-verify-source

```diff
@@ -0,0 +1,392 @@
+    echo "ERROR: KEYRING_DIR_GIT is not set!" >&2
+    exit 1
+}
+
+# Default GIT suffix
+[ -z "$GIT_SUFFIX" ] && GIT_SUFFIX='.git'
+
+# Default GIT branch
+[ -z "$BRANCH" ] && BRANCH='master'
+
+# Allow to override URL for which COMPONENT is cloned
+if [ -n "${GIT_URL-}" ]; then
+    GIT_URL="$GIT_URL"
+elif [ -n "${GIT_BASEURL}" ] && [ -n "${GIT_PREFIX}" ]; then
+    GIT_URL=$GIT_BASEURL/$GIT_PREFIX$COMPONENT$GIT_SUFFIX
+else
+    echo 'ERROR: Cannot determine how to clone repository!' >&2
+    exit 1
+fi
+
+# Override GIT_URL with GIT_REMOTE if given
+[ -n "${GIT_REMOTE=}" ] && GIT_URL=$GIT_REMOTE
+
+# Default values
+: "${IGNORE_MISSING=0}"
+: "${GIT_CLONE_FAST=0}"
+
+# Define common GIT options
+GIT_OPTIONS=
+if [ "$GIT_CLONE_FAST" = "1" ]; then
+    GIT_OPTIONS+="--depth=1"
+fi
+
+# Sanity check on BRANCH and REPO
+if ! [[ "$BRANCH" =~ ^[A-Za-z][A-Za-z0-9._-]+$ ]]; then
+    printf 'Invalid branch %q\n' "$BRANCH" >&2
+    exit 1
+elif ! [[ "$REPO" =~ ^[A-Za-z][A-Za-z0-9-]*$ ]]; then
+    printf 'Invalid repository %q\n' "$REPO" >&2
+    exit 1
+fi
+
+trap 'exit_clean' 0 1 2 3 6 15
+
+fresh_clone=false
+if [ -d "$REPO" ] && [ "${CLEAN-}" != '1' ]; then
+    cd "$REPO"
+    if [ "$GIT_CLONE_FAST" != "1" ] && [ "$(git rev-parse --is-shallow-repository)" == "true" ]; then
+        GIT_OPTIONS+="--unshallow"
+    fi
+    print_headers
+    if ! git fetch $GIT_OPTIONS -q --tags -- "$GIT_URL" "$BRANCH"; then
+        if [ "$IGNORE_MISSING" = "1" ]; then exit 0; else exit 1; fi
+    fi
+    VERIFY_REF=FETCH_HEAD fresh_clone=false
+    # shellcheck disable=SC2103
+    cd - >/dev/null
+else
+    rm -rf "$REPO"
+    print_headers
+    if ! git clone $GIT_OPTIONS -n -q -b "$BRANCH" "$GIT_URL" "$REPO"; then
+        if [ "$IGNORE_MISSING" == "1" ]; then exit 0; else exit 1; fi
+    fi
+    VERIFY_REF=HEAD fresh_clone=:
+fi
+
+export CHECK=signed-tag
+
+verify=true
+if [ "$INSECURE_SKIP_CHECKING" == "1" ]; then
+    verify=false
+elif [ "$LESS_SECURE_SIGNED_COMMITS_SUFFICIENT" == "1" ]; then
+    CHECK=signed-tag-or-commit
+fi
+
+VERIFY_REF=$(git -C "$REPO" rev-parse -q --verify "$VERIFY_REF") || exit
+
+if [ "$verify" = 'false' ]; then
+    echo -e '\033[1;31m--> NOT verifying tags\033[0;0m'
+else
+    if [[ $CHECK == 'signed-tag-or-commit' ]]; then
+        echo "--> Verifying tags or commits..."
+    else
+        echo "--> Verifying tags..."
+    fi
+
+    GNUPGHOME="$(readlink -m "$KEYRING_DIR_GIT")"
+    export GNUPGHOME
+    if [ ! -d "$GNUPGHOME" ]; then
+        mkdir -p "$GNUPGHOME"
+        chmod 700 "$GNUPGHOME"
+        gpg --import "$KEYS_DIR/qubes-developers-keys.asc"
+        # Trust Qubes Master Signing Key
+        echo '427F11FD0FAA4B080123F01CDDFA1A3E36879494:6:' | gpg --import-ownertrust
+    fi
+    if [ "$KEYS_DIR/qubes-developers-keys.asc" -nt "$GNUPGHOME/pubring.gpg" ]; then
+        gpg --import "$KEYS_DIR/qubes-developers-keys.asc"
+        touch "$GNUPGHOME/pubring.gpg"
+    fi
+    for keyid in ${MAINTAINERS[*]}; do
```

Comment:
```suggestion
    for keyid in "${MAINTAINERS[@]}"; do
```

This does not matter in practice because `MAINTAINERS` has a restricted character set, but it is still best practice.

Reply:
Already fixed.